PyInstaller/building/osx.py

```diff
@@ -137,20 +140,16 @@ def assemble(self):
                            "CFBundlePackageType": "APPL",
                            "CFBundleShortVersionString": self.version,
 
-                           # Setting this to 1 will cause Mac OS X *not* to show
-                           # a dock icon for the PyInstaller process which
-                           # decompresses the real executable's contents. As a
-                           # side effect, the main application doesn't get one
-                           # as well, but at startup time the loader will take
-                           # care of transforming the process type.
-                           "LSBackgroundOnly": False,
-
                            }
 
         # Merge info_plist settings from spec file
         if isinstance(self.info_plist, dict) and self.info_plist:
             info_plist_dict.update(self.info_plist)
 
+        # Setting EXE console=True implicitely sets LSBackgroundOnly to True
+        if self.console:
+            info_plist_dict['LSBackgroundOnly'] = True
```

Comment:
If we move this in front of "Merge info_plist settings from spec file", this would allow users to overwrite the `LSBackgroundOnly` if needed for some reason. What do you think?

Reply:
Nice catch, change applied.